numpy/core/src/multiarray/methods.c

```diff
@@ -988,13 +1053,15 @@ array_ufunc(PyArrayObject *self, PyObject *args, PyObject *kwds)
         return NULL;
     }
     /* ndarray cannot handle overrides itself */
-    num_override_args = PyUFunc_WithOverride(normal_args, kwds, NULL, NULL);
-    if (num_override_args == -1) {
-        return NULL;
-    }
-    if (num_override_args) {
-        result = Py_NotImplemented;
-        Py_INCREF(Py_NotImplemented);
+    has_override = check_override(normal_args, kwds);
+    if (has_override) {
+        if (has_override == -1) {
```

Comment:
`has_override` shouldn't be an `npy_bool` if it contains -1

Reply:
Indeed. Leftover. I do wonder to what extent I should even check that `args` is a tuple and `kwds` a dict; one can presumably only get here via python, so this should be OK. What do you think? remove those tests?

Reply:
Could write them as `assert`s

Reply:
Good idea, done in `array_ufunc` itself.